packages/blockchain/src/index.ts

```diff
@@ -641,6 +646,186 @@ export default class Blockchain implements BlockchainInterface {
     })
   }
 
+  /**
+   * Validates a block header, throwing if invalid. It is being validated against the reported `parentHash`.
+   * It verifies the current block against the `parentHash`:
+   * - The `parentHash` is part of the blockchain (it is a valid header)
+   * - Current block number is parent block number + 1
+   * - Current block has a strictly higher timestamp
+   * - Additional PoW checks ->
+   *   - Current block has valid difficulty and gas limit
+   *   - In case that the header is an uncle header, it should not be too old or young in the chain.
+   * - Additional PoA clique checks ->
+   *   - Checks on coinbase and mixHash
+   *   - Current block has a timestamp diff greater or equal to PERIOD
+   *   - Current block has difficulty correctly marked as INTURN or NOTURN
+   * @param header - header to be validated
+   * @param height - If this is an uncle header, this is the height of the block that is including it
+   */
+  public async validateHeader(header: BlockHeader, height?: bigint) {
+    if (header.isGenesis()) {
+      return
+    }
+    const parentHeader = (await this.getBlock(header.parentHash)).header
+    if (!parentHeader) {
+      const msg = header._errorMsg('could not find parent header')
+      throw new Error(msg)
+    }
+
+    const { number } = header
+    if (number !== parentHeader.number + BigInt(1)) {
+      const msg = header._errorMsg('invalid number')
+      throw new Error(msg)
+    }
+
+    if (header.timestamp <= parentHeader.timestamp) {
+      const msg = header._errorMsg('invalid timestamp')
+      throw new Error(msg)
+    }
+
+    if (this._common.consensusAlgorithm() === ConsensusAlgorithm.Clique) {
+      const period = (this._common.consensusConfig() as CliqueConfig).period
+      // Timestamp diff between blocks is lower than PERIOD (clique)
+      if (parentHeader.timestamp + BigInt(period) > header.timestamp) {
+        const msg = header._errorMsg('invalid timestamp diff (lower than period)')
+        throw new Error(msg)
+      }
+      // Validate clique difficulty
+      await this.consensus.validateDifficulty(header)
+    }
```

Comment:
Now we have this *one* consensus-related check back in here which is unlucky. Wonder if there is a good place in `consensus` we can safely move.

Maybe in the `clique.validate()` method as well? 🤔 Might need a double check if the semantic result would be somewhat matching/equivalent.

But we should really target to remove here.